File: API/Entities/Product.cs
namespace API.Entities
{
    public class Product
    {
           public int Id { get; set; }        
           public string Name { get; set; }        
           public string Description { get; set; }        
           public long Price { get; set; }        //putting long because on the UI I'll add the decimals ex: 10000 in the dB will be 100.00 dollars
           public string PictureUrl { get; set; }        
           public string Type { get; set; }        
           public string Brand { get; set; }        
           public int QuantityInStock { get; set; }        
    }
}

File: client/src/app/models/product.ts
export interface Product{
    id: number
    name: string
    description: string
    price: number
    pictureUrl: string
    type?: string
    brand: string
    quantityInStock?: number
}

// first step to add product parameters to the redux state. the goal will be to append query params to the /api/products(...) endpoint
export interface ProductParams{
    orderBy: string;
    searchTerm?: string;
    types: string[];
    brands: string[];
    pageNumber: number;
    pageSize: number;
}